packages/flutter/lib/src/widgets/preferred_size.dart

```diff
@@ -0,0 +1,47 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+
+import 'basic.dart';
+import 'framework.dart';
+
+/// Return the size this widget would prefer if it were otherwise unconstrained.
```

Comment:
I didn't get confused. I think this should start off as "Interface for Widgets that can return the size this widget would prefer if...".